pytests/tests/test_check_update.py

```diff
@@ -6,16 +6,30 @@
 # of the License are located in the COPYING file of this distribution.
 #
 
+import os
 import pytest
+import shutil
+
+WORKDIR = '/root/test_config'
+TEST_CONF_FILE = 'tdnf.conf'
+TEST_CONF_PATH = os.path.join(WORKDIR, TEST_CONF_FILE)
 
 
 @pytest.fixture(scope='module', autouse=True)
 def setup_test(utils):
+    tdnf_conf = os.path.join(utils.config['repo_path'], 'tdnf.conf')
+
+    utils.makedirs(WORKDIR)
+    shutil.copy(tdnf_conf, TEST_CONF_PATH)
+
+    utils.edit_config({'dnf_check_update_compat': '1'}, section='main', filename=TEST_CONF_PATH)
```

Comment:
I see this is always enabled in all tests. We need to test with the setting enabled and with the setting disabled.

You can modify `test_check_update_multi_version_package` to check that it still returns 0 when updates are available, and do the setting `dnf_check_update_compat=1` only for `test_check_update_dnf_compat()`. You can also add a test that with the setting it still returns 0 when no updates are available.

Reply:
So I based this on test_config.py, and I believe even though it appears it's enabled for all tests, it is only used when we use `--config`, `TEST_CONF_PATH` otherwise it uses the config file without the `dnf_check_update_compat=` that was added.

Reply:
Oh, yes, you are correct. I missed that.